internal/service/encryptionatrest/model_encryption_at_rest_test.go

```diff
@@ -0,0 +1,273 @@
+	}
+	TfAwsKmsConfigModel = encryptionatrest.TfAwsKmsConfigModel{
+		Enabled:             types.BoolValue(enabled),
+		CustomerMasterKeyID: types.StringValue(customerMasterKeyID),
+		Region:              types.StringValue(region),
+		AccessKeyID:         types.StringValue(accessKeyID),
+		SecretAccessKey:     types.StringValue(secretAccessKey),
+		RoleID:              types.StringValue(roleID),
+	}
+	AzureKeyVault = &admin.AzureKeyVault{
+		Enabled:           &enabled,
+		ClientID:          &clientID,
+		AzureEnvironment:  &azureEnvironment,
+		SubscriptionID:    &subscriptionID,
+		ResourceGroupName: &resourceGroupName,
+		KeyVaultName:      &keyVaultName,
+		KeyIdentifier:     &keyIdentifier,
+		TenantID:          &tenantID,
+		Secret:            &secret,
+	}
+	TfAzureKeyVaultConfigModel = encryptionatrest.TfAzureKeyVaultConfigModel{
+		Enabled:           types.BoolValue(enabled),
+		ClientID:          types.StringValue(clientID),
+		AzureEnvironment:  types.StringValue(azureEnvironment),
+		SubscriptionID:    types.StringValue(subscriptionID),
+		ResourceGroupName: types.StringValue(resourceGroupName),
+		KeyVaultName:      types.StringValue(keyVaultName),
+		KeyIdentifier:     types.StringValue(keyIdentifier),
+		TenantID:          types.StringValue(tenantID),
+		Secret:            types.StringValue(secret),
+	}
+	GoogleCloudKMS = &admin.GoogleCloudKMS{
+		Enabled:              &enabled,
+		KeyVersionResourceID: &keyVersionResourceID,
+		ServiceAccountKey:    &serviceAccountKey,
+	}
+	TfGcpKmsConfigModel = encryptionatrest.TfGcpKmsConfigModel{
+		Enabled:              types.BoolValue(enabled),
+		KeyVersionResourceID: types.StringValue(keyVersionResourceID),
+		ServiceAccountKey:    types.StringValue(serviceAccountKey),
+	}
+	EncryptionAtRest = &admin.EncryptionAtRest{
+		AwsKms:         AWSKMSConfiguration,
+		AzureKeyVault:  AzureKeyVault,
+		GoogleCloudKms: GoogleCloudKMS,
+	}
+)
+
+func TestNewTfEncryptionAtRestRSModel(t *testing.T) {
+	testCases := []struct {
+		expectedResult *encryptionatrest.TfEncryptionAtRestRSModel
+		sdkModel       *admin.EncryptionAtRest
+		name           string
+	}{
+		{
+			name:     "Success NewTFAwsKmsConfig",
+			sdkModel: EncryptionAtRest,
+			expectedResult: &encryptionatrest.TfEncryptionAtRestRSModel{
+				ID:                   types.StringValue(projectID),
+				ProjectID:            types.StringValue(projectID),
+				AwsKmsConfig:         []encryptionatrest.TfAwsKmsConfigModel{TfAwsKmsConfigModel},
+				AzureKeyVaultConfig:  []encryptionatrest.TfAzureKeyVaultConfigModel{TfAzureKeyVaultConfigModel},
+				GoogleCloudKmsConfig: []encryptionatrest.TfGcpKmsConfigModel{TfGcpKmsConfigModel},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel := encryptionatrest.NewTfEncryptionAtRestRSModel(context.Background(), projectID, tc.sdkModel)
+			assert.Equal(t, tc.expectedResult, resultModel, "created terraform model did not match expected output")
```

Comment:
knitpit: I'm not a big fan of assert messages except if they add something important. I think in this case it can be enough with the default assert messaging.

Reply:
Makes sense, will change it!